src/Common/tests/System/AssertExtensions.cs

```diff
@@ -50,5 +50,26 @@ public static T Throws<T>(string paramName, Func<object> testCode)
 
             return exception;
         }
+
+        public static void Throws(Type netCoreExceptionType, string netCoreParamName, Type netFxExceptionType, string netFxParamName, Action action)
```

Comment:
I'm having this issue as well so it might be worth adding this extension method, I was thinking also of adding a ThrowsAny<T> extension method that takes one and two param names.

I can add @stephentoub's suggestion which makes sense and then you can update this PR to use that extension method, or the other way around, whichever you'd like.